### ISSUE TITLE
Refetch does not refetch with SSR

### ISSUE THREAD

**Issue by nickluger** [apollographql/apollo-client#3248]
**TL;DR:**  If my app is SSRenderd and my cache policy on a query is `cacheFirst`, subsequent calls to `refetch` do not lead to expected server requests but are resolved from the cache. 

Setting the policy to `cache-and-network` solves the problem but causes others:
https://github.com/apollographql/apollo-client/issues/2119

Manually resetting the store on the client solves the problem, too:
`__APOLLO_CLIENT__.resetStore()`

How to reproduce?

1. Make a `<Query/>` on a certain component
 `<Query query={MY_POSTS_QUERY} fetchPolicy='cache-first'>`

2. Do some SSR stuff
```js
export default function initApollo (initialState, userDataOnServer) {
  if (isServer()) {
    return create(initialState, userDataOnServer)
  }
  // Reuse client on the client-side
  if (!apolloClient) {
    apolloClient = create(initialState)
  }
  return apolloClient
}
// ...
function create (initialState, userDataOnServer) {
  return new ApolloClient({
    connectToDevTools: !isServer(),
    ssrMode: isServer(), // Disables forceFetch on the server (so queries are only run once)
    link: from([errorLink, createAuthLink(userDataOnServer), httpLink]),
    cache: new InMemoryCache().restore(initialState || {})
  })
}

// ...
await getDataFromTree(
            <ApolloProvider client={apollo}>
              <ComposedComponent url={url} {...composedInitialProps} />
            </ApolloProvider>
          )
```

3. Result is rendered correctly on the server and HTML is returned with the Apollo cache pre-filled.

4. **Calling refetch does reach to the network.** Calling refetch with **different variables** does not work neither.

Is this behaviour intended?

**Comment by wesbos** [comment]
Man am I glad to see this issue - been trying for a few hours to figure this one out. 

Any solution to this?

**Comment by evans** [comment]
Thanks for opening the issue! This should be solved with https://github.com/apollographql/react-apollo/pull/1901 🎉

**Comment by hwillson** [comment]
Fixed by https://github.com/apollographql/react-apollo/pull/1901. Thanks!
